contracts/apps/disputable/IAgreement.sol

```diff
@@ -0,0 +1,37 @@
+/*
+ * SPDX-License-Identitifer:    MIT
+ */
+
+pragma solidity ^0.4.24;
+
+import "../../acl/IACLOracle.sol";
+import "../../lib/token/ERC20.sol";
+import "../../lib/arbitration/IArbitrable.sol";
+
+
+contract IAgreement is IArbitrable, IACLOracle {
+    function sign() external;
+
+    function newAction(uint256 _disputableActionId, bytes _context, address _submitter, uint64 _lifetime) external returns (uint256);
+
+    function closeAction(uint256 _actionId) external;
+
+    function challengeAction(uint256 _actionId, uint256 _settlementOffer, bool _finishedSubmittingEvidence, bytes _context) external;
+
+    function settle(uint256 _actionId) external;
+
+    function disputeAction(uint256 _actionId, bool _finishedSubmittingEvidence) external;
+
+    function register(
+        address _disputable,
+        ERC20 _collateralToken,
+        uint256 _actionAmount,
+        uint256 _challengeAmount,
+        uint64 _challengeDuration
+    )
+        external;
+
+    function unregister(address _disputable) external;
+
+    function canProceed(uint256 _actionId) external view returns (bool);
```

Comment:
What do you think about also including a getter for getting the disputable's settings info, like the one we've already implemented in [`getDisputableInfo()`](https://github.com/aragon/aragon-apps/blob/implement_agreements_app/apps/agreement/contracts/Agreement.sol#L587)?

Not sure about the `getDisputableInfo()` name, but I think it'll be useful as an expected part of the interface of any Agreement.

Reply:
And perhaps we should expose `getChallenge()` as well?

Thinking about this from the perspective of a subgraph/reducer running with only the Voting app and `IArbitrable` ABI in mind. Ideally on each event emitted from one of the `IDisputable`'s callbacks, we are able to reach in and fetch any updated information from the Agreement.

Reply:
> What do you think about also including a getter for getting the disputable's settings info

Not sure I follow what's the information you expect to fetch from this new getter, is it the disputable info or the disputable's collateral requirement?

Reply:
> And perhaps we should expose `getChallenge()` as well?

The problem with this one is that forces you to basically have a lot of information for each challenge, I think adding this one will force us to add the rest of the getters, otherwise it won't make a lot of sense.

Reply:
> Not sure I follow what's the information you expect to fetch from this new getter, is it the disputable info or the disputable's collateral requirement?

This would be the collateral requirements, since these can change per action.

> The problem with this one is that forces you to basically have a lot of information for each challenge

Let's discuss!